app/stores/interactive-filters.tsx

```diff
@@ -141,80 +142,62 @@ type InteractiveFiltersContextValue = [
   StoreApi<State>,
 ];
 
-export type SharedFilter = {
-  type: "timeRange" | "dataFilters";
-  value?: NonNullable<InteractiveFiltersConfig>["timeRange"];
-  iri: string;
-};
+export type SharedFilter =
+  | ({
+      type: "timeRange";
+    } & NonNullable<InteractiveFiltersConfig>["timeRange"])
+  | ({
+      type: "dataFilters";
+    } & NonNullable<InteractiveFiltersConfig>["dataFilters"]);
 
-type SharedFilters = SharedFilter[];
+export type PotentialSharedFilter = Pick<SharedFilter, "type" | "componentIri">;
 
 const InteractiveFiltersContext = createContext<
   | {
       stores: Record<ChartConfig["key"], InteractiveFiltersContextValue>;
-      sharedFilters: SharedFilters;
+      potentialSharedFilters: PotentialSharedFilter[];
+      sharedFilters: SharedFilter[];
     }
   | undefined
 >(undefined);
 
 /**
  * Returns filters that are shared across multiple charts.
  */
-export const getSharedFilters = (
-  state: ConfiguratorStateWithChartConfigs
-): SharedFilters => {
-  const chartConfigs = state.chartConfigs;
-
-  // We show shared filters only when the chart is not being configured
-  if (state.state === "CONFIGURING_CHART") {
-    return [];
-  }
-
-  // Shared filters are only relevant for dashboard layout
-  if (state.layout.type !== "dashboard") {
-    return [];
-  }
-
-  if (chartConfigs.length < 2) {
-    return [];
-  }
-
-  const allActiveFilters = chartConfigs.flatMap((c) => {
-    const interactiveFiltersConfig = c.interactiveFiltersConfig;
-    if (!interactiveFiltersConfig) {
-      return [];
-    }
-
-    const { timeRange } = interactiveFiltersConfig;
-    return [{ type: "timeRange" as const, value: timeRange }].filter(
-      (x) => x.value.active
+export const getPotentialSharedFilters = (
+  chartConfigs: ChartConfig[]
+): PotentialSharedFilter[] => {
+  const temporalDimensions = chartConfigs.flatMap((c) => {
+    const chartSpec = getChartSpec(c);
+    const temporalEncodings = chartSpec.encodings.filter((x) =>
+      x.componentTypes.some((x) => x === "TemporalDimension")
```

Comment:
Wondering if `TemporalEntityDimension` wouldn't also work "out of the box", as it looks like it has a `timeUnit` just like `TemporalDimension`?

Reply:
I created an issue for that : https://github.com/visualize-admin/visualization-tool/issues/1555

Reply:
Thanks 🙇‍♂️